pkg/utils/validator/clusterresourceplacement.go

```diff
@@ -346,3 +364,44 @@ func validateRolloutStrategy(rolloutStrategy placementv1beta1.RolloutStrategy) e
 
 	return apiErrors.NewAggregate(allErr)
 }
+
+// validatePropertySelector validates the property selector
+func validatePropertySelector(propertySelector *placementv1beta1.PropertySelector) error {
+	if reflect.TypeOf(propertySelector.MatchExpressions).Kind() == reflect.Slice && reflect.TypeOf(propertySelector.MatchExpressions).Elem().Name() == "PropertySelectorRequirement" {
+		// selector.MatchExpressions is of type []PropertySelectorRequirement
+		return validatePropertySelectorRequirements(propertySelector.MatchExpressions)
+	}
+	// MatchExpression is not of type []PropertySelectorRequirement
+	return fmt.Errorf("invalid MatchExpressions type %T, should be []PropertySelectorRequirement", reflect.TypeOf(propertySelector.MatchExpressions))
+}
+
+func validatePropertySelectorRequirements(propertySelectorRequirement []placementv1beta1.PropertySelectorRequirement) error {
+	allErr := make([]error, 0)
+	for _, propertySelectorRequirement := range propertySelectorRequirement {
+		// Validate the property name is a Kubernetes label name
+		errs := validation.IsQualifiedName(propertySelectorRequirement.Name)
+		if errs != nil {
+			allErr = append(allErr, fmt.Errorf("invalid property name %s: %v", propertySelectorRequirement.Name, errs))
+		}
+
+		validOperators := map[placementv1beta1.PropertySelectorOperator]bool{
+			placementv1beta1.PropertySelectorGreaterThan:          true,
+			placementv1beta1.PropertySelectorGreaterThanOrEqualTo: true,
+			placementv1beta1.PropertySelectorLessThan:             true,
+			placementv1beta1.PropertySelectorLessThanOrEqualTo:    true,
+			placementv1beta1.PropertySelectorEqualTo:              true,
+			placementv1beta1.PropertySelectorNotEqualTo:           true}
+
+		if validOperators[propertySelectorRequirement.Operator] && len(propertySelectorRequirement.Values) != 1 {
```

Comment:
Hi Britania! Just a nit: would you mind adding a comment here explaining that for the current list of supported operators, the expected length of values is always 1?

Reply:
Same for the quantity check below -> i.e., currently we expect the value to be a valid resource.Quantity.